tests/test_providers.py

```diff
@@ -839,3 +846,39 @@ def parse_(self, response: DummyResponse, page1: FirstPage, page2: SecondPage):
     assert type(item["page2"].http_response) is HttpResponse
     assert type(item["page2"].response) is AnyResponse
     assert type(item["page2"].response.response) is BrowserResponse
+
+
+@ensureDeferred
+async def test_screenshot():
+    settings = create_scrapy_settings()
+    settings["SCRAPY_POET_PROVIDERS"] = {ZyteApiProvider: 0}
+    crawler = get_crawler(settings_dict=settings)
+    injection_mw = InjectionMiddleware.from_crawler(crawler)
```

Comment:
Would it make sense to test this feature using mockserver instead? I.e. add screenshot feature to the mockserver, and then just check that the dependency works as expected?

Reply:
I made an effort to avoid it due to https://github.com/scrapy/scrapy/issues/6255. I also feel like off-mockserver approaches give more granular control, although they tend to be less readable and more likely to break with future upstream API changes.

Shall I switch it to mockserver nonetheless?

Reply:
Mockserver doesn't have to be slow: the startup time can be amortized using pytest fixtures bound to session, and other things should be pretty fast. I recall it was quite fast e.g. in Splash. Haven't checked the Scrapy issue.

Probably that's personal, but I find low-level mock tests to often be an obstacle (instead of a help) in case of refactoring. Also, they don't actually show that the feature works from the user point of view.